app/src/main/java/de/westnordost/streetcomplete/quests/surface/DetailRoadSurface.kt

```diff
@@ -0,0 +1,101 @@
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.elementgeometry.ElementGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.OverpassMapDataAndGeometryApi
+import de.westnordost.streetcomplete.data.osm.osmquest.OsmElementQuestType
+import de.westnordost.streetcomplete.data.tagfilters.FiltersParser
+import de.westnordost.streetcomplete.data.tagfilters.getQuestPrintStatement
+import de.westnordost.streetcomplete.data.tagfilters.toGlobalOverpassBBox
+
+
+class DetailRoadSurface(private val overpassMapDataApi: OverpassMapDataAndGeometryApi) : OsmElementQuestType<DetailSurfaceAnswer> {
+    override val commitMessage = "Add more detailed surfaces"
+    override val wikiLink = "Key:surface"
+    override val icon = R.drawable.ic_quest_street_surface_detail
+
+    override fun getTitle(tags: Map<String, String>): Int {
+        val hasName = tags.containsKey("name")
+        val isSquare = tags["area"] == "yes"
+
+        return if (hasName) {
+            if (isSquare)
+                R.string.ic_quest_surface_detailed_square_name_title
+            else
+                R.string.ic_quest_surface_detailed_name_title
+        } else {
+            if (isSquare)
+                R.string.ic_quest_surface_detailed_square_title
+            else
+                R.string.ic_quest_surface_detailed_title
+        }
+    }
+
+    override fun createForm() = DetailRoadSurfaceForm()
+
+    override fun download(bbox: BoundingBox, handler: (element: Element, geometry: ElementGeometry?) -> Unit): Boolean {
+        return overpassMapDataApi.query(getOverpassQuery(bbox), handler)
+    }
+
+    override fun isApplicableTo(element: Element): Boolean? {
+        if(!REQUIRED_MINIMAL_MATCH_TFE.matches(element)) {
+            return false;
+        }
+        element.tags.forEach {
+            if(it.key.contains("surface:")) {
+                return false;
+            }
+            if(it.key.contains(":surface")) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private fun getOverpassQuery(bbox: BoundingBox) =
+        bbox.toGlobalOverpassBBox() + "\n" + """
+
+          way[surface~"^(${UNDETAILED_SURFACE_TAG_MATCH})${'$'}"][segregated!="yes"][highway ~ "^${ HIGHWAY_TAG_MATCH }${'$'}"] -> .surface_without_detail;
+          // https://taginfo.openstreetmap.org//search?q=%3Asurface
+          // https://taginfo.openstreetmap.org//search?q=surface:
+          way[surface~"^(${UNDETAILED_SURFACE_TAG_MATCH})${'$'}"][~"(:surface|surface:)" ~ "."] -> .extra_tags;
+          way.surface_without_detail[surface ~ "^(${UNDETAILED_SURFACE_TAG_MATCH})${'$'}"][access !~ "^(private|no)${'$'}"] -> .not_private;
+          way.surface_without_detail[surface ~ "^(${UNDETAILED_SURFACE_TAG_MATCH})${'$'}"][foot][foot !~ "^(private|no)${'$'}"] -> .foot_access;
+          ((.not_private; .foot_access;); - .extra_tags;);
```

Comment:
All this `(:surface|surface:)` tags are really rare, so skipping them altogether is not a significant impact. And handling them is tricky (for example what should be done with `source:surface` while `surface` is edited? Removed? But what if it applies to `cycleway:right:surface`). So I decided to not think about handling them and just skip rare cases where such tags are present.

It also skips problem of people already specifying detailed surface using various tagging schemes (from relatively common to absurdly rare ones that I used in https://www.openstreetmap.org/way/141471022 ) - and I wanted to avoid "pointless" and pointless questions. It is also deliberately done to avoid irritating people tagging it somehow. Maybe I am too careful here.

Reply:
You think `surface:left` and `right` is absurd? Sounds fine to me, actually.

Anyway, I think at least after the sidewalk- and cycleway-surface quest are implemented, `cycleway:left/right/both:surface` etc. will be quite common.

I think:
- `source:surface` could be either deleted or ignored (better deleted)
- if `note:surface` (or `description:surface`.. only 95 usages) is defined, do not ask the question
- and the rest ignored

Reply:
By the way, maybe it is less code and easier to maintain if you used the tag filter expression to download but do the additional check in overriding `download`

Reply:
Looking at https://taginfo.openstreetmap.org/keys/description%3Asurface#values it seems that including them should not be a problem and this key is really rare anyway.

Thinking about it again - adding `surface:note` with explanation to cases known to me would not be a problem anyway.

I will switch exclusion to just `surface:note`